igc.cc

```diff
@@ -44,11 +44,14 @@
 #include <QTime>                // for operator<, operator==, QTime
 #include <Qt>                   // for UTC, SkipEmptyParts
 #include <QtGlobal>             // for foreach, qPrintable
+#include <QStringRef>           // for substring
```

Comment:
I'm not familiar with Qt, QStringRef does the job well without creating a new QString.

As you mentioned below, does QByteArray perform the same function (i.e. I want to use a pointer to an existing string, rather than creating a new one)?

Reply:
The bigger save is QByteArray doesn't do an 8 bit to UTF-16 conversion (and back).  

I would advice against worrying about the copy.  Eventually we will use QStringView more, but at the moment we are supporting both Qt5 and Qt6.  QByteArray::fromRawData can avoid the copy, but can lead to other issues so we haven't used it.